### PR TITLE
chore: pnnsGroup refactoring

### PR DESCRIPTION
**Pull request by monsieurtanuki**
### What
- Some refactoring around pnns groups, that were using a slightly obscure syntax that we can avoid now due to dart syntax evolution.

### Impacted files
* `pnns_group2_filter.dart`: minor refactoring
* `pnns_groups.dart`: used a clearer syntax

### CODE REVIEW

**Comment by monsieurtanuki** [comment]
> Error: Unable to resolve action `codecov/codecov-action@v4`, unable to find version `v4`

@teolemon Looks familiar?

**Comment by teolemon** [comment]
ah snap, I forgot about this one, and merged it @monsieurtanuki

File: lib/src/utils/pnns_groups.dart [hunk 1]
```diff
@@ -1,233 +1,276 @@
...
+  FRUIT_NECTARS(
+    offTag: 'fruit-nectars',
+    name: 'Fruits nectars',
+  ),
+  TEAS_AND_HERBAL_TEAS_AND_COFFEES(
+    offTag: 'teas-and-herbal-teas-and-coffees',
+    name: 'Teas and herbal teas and coffees',
+  ),
+
+  BREAKFAST_CEREALS(
```
Comment (M123-dev) [inline lib/src/utils/pnns_groups.dart:137]: Why are there sometimes new lines between the subgroups and sometimes not

**Review by M123-dev — APPROVED** [review]
Thanks @monsieurtanuki, just one minor inconsistency but thats nothing to block this PR on

File: lib/src/utils/pnns_groups.dart (hunk 1 above)
Comment (monsieurtanuki) [reply lib/src/utils/pnns_groups.dart:137]: > Why are there sometimes new lines between the subgroups and sometimes not

The usual question: the empty lines were there the way it was written previously (in that specific case, between level-1 groups of level-2 groups).
It doesn't seem to trigger lints warning so why not. Of course we could get rid of those useless empty lines.

Thank you @M123-dev for your review!

**Pull request merged**